spec/features/user_returns_checked_out_book_spec.rb

```diff
@@ -0,0 +1,24 @@
+require "rails_helper"
+
+feature "User returns only books she borrowed from the library" do
+  scenario "successfully" do
+    book_1 = create(:book)
+    patron_1 = create(:patron)
+    sign_in_and_check_out_book(patron_1)
```

Comment:
Yeah. I don't think I'd bother extracting that helper method at all. `sign_in_and_check_out` seem like two different steps that I'd be happy to see repeated as often as they need to be.

```ruby
visit books_path(as: user)
click_on "Check out"
```

seems quite readable to me.